hyperspy/roi.py

```diff
@@ -322,11 +324,15 @@ def interactive(self, signal, navigation_signal="same", out=None,
             If not None, it will use 'out' as the output instead of returning
             a new Signal.
         """
-        if navigation_signal == "same":
+        if isinstance(navigation_signal, basestring) and navigation_signal == "same":
             navigation_signal = signal
         if navigation_signal is not None:
             if navigation_signal not in self.signal_map:
                 self.add_widget(navigation_signal)
+        if self.update not in signal.axes_manager.events.any_axis_changed.connected:
+            signal.axes_manager.events.any_axis_changed.connect(
+                self.update,
+                [])
```

Comment:
This connection can prevent the ROI from being garbage collected.

```python
s = hs.signals.Spectrum(np.random.rand(10, 10, 10))
r = hs.roi.Point2DROI(0,0)
sr = r.interactive(s, axes=s.axes_manager.navigation_axes)
r.remove_widget(s)
import gc
gc.collect()
del sr
del r
assert gc.collect() >= 2
# Here, both r and sr is kept alive by event connections?
# sr is kept alive because of issues with hs.interactive code
# r is kept alive because of this connection (and others?)
```